packages/event-handler/tests/unit/rest/RouteHandlerRegistry.test.ts

```diff
@@ -0,0 +1,222 @@
+    it('returns empty array for non-existent method', () => {
+      // Prepare
+      const mockLogger = { debug: vi.fn(), warn: vi.fn(), error: vi.fn() };
+      const registry = new RouteHandlerRegistry({ logger: mockLogger });
+
+      // Act & Assert
+      expect(registry.getRoutesByMethod('GET')).toEqual([]);
+      expect(registry.getRoutesByMethod('POST')).toEqual([]);
+    });
+
+    it.each([
+      { method: HttpVerbs.GET },
+      { method: HttpVerbs.POST },
+      { method: HttpVerbs.PUT },
+      { method: HttpVerbs.DELETE },
+    ])('returns routes for $method method', ({ method }) => {
+      // Prepare
+      const mockLogger = { debug: vi.fn(), warn: vi.fn(), error: vi.fn() };
+      const registry = new RouteHandlerRegistry({ logger: mockLogger });
+      const handler = () => 'test';
+
+      const route1 = new Route(method, '/users', handler);
+      const route2 = new Route(method, '/posts', handler);
+      const otherMethodRoute = new Route(HttpVerbs.PATCH, '/other', handler);
+
+      // Act
+      registry.register(route1);
+      registry.register(route2);
+      registry.register(otherMethodRoute);
+
+      // Assert
+      const routes = registry.getRoutesByMethod(method);
+      expect(routes).toHaveLength(2);
+      expect(routes).toContain(route1);
+      expect(routes).toContain(route2);
+      expect(routes).not.toContain(otherMethodRoute);
+    });
+
+    it('handles case-insensitive method lookup', () => {
+      // Prepare
+      const mockLogger = { debug: vi.fn(), warn: vi.fn(), error: vi.fn() };
+      const registry = new RouteHandlerRegistry({ logger: mockLogger });
+      const handler = () => 'test';
+
+      const getRoute = new Route(HttpVerbs.GET, '/users', handler);
+
+      // Act
+      registry.register(getRoute);
+
+      // Assert
+      expect(registry.getRoutesByMethod('get')).toContain(getRoute);
+      expect(registry.getRoutesByMethod('GET')).toContain(getRoute);
+      expect(registry.getRoutesByMethod('Get')).toContain(getRoute);
+    });
+  });
+
+  describe('getAllRoutes', () => {
+    it('returns empty array for empty registry', () => {
+      // Prepare
+      const mockLogger = { debug: vi.fn(), warn: vi.fn(), error: vi.fn() };
+      const registry = new RouteHandlerRegistry({ logger: mockLogger });
+
+      // Act & Assert
+      expect(registry.getAllRoutes()).toEqual([]);
+    });
+
+    it('returns all registered routes', () => {
+      // Prepare
+      const mockLogger = { debug: vi.fn(), warn: vi.fn(), error: vi.fn() };
+      const registry = new RouteHandlerRegistry({ logger: mockLogger });
+      const handler = () => 'test';
+
+      const route1 = new Route(HttpVerbs.GET, '/users', handler);
+      const route2 = new Route(HttpVerbs.POST, '/users', handler);
+      const route3 = new Route(HttpVerbs.GET, '/posts', handler);
+
+      // Act
+      registry.register(route1);
+      registry.register(route2);
+      registry.register(route3);
+
+      // Assert
```

Comment:
```suggestion
      // Assess
```

`s/Assert/Assess/` in the file